mautic/mautic
# [DPMMA-1866] Map with e-mail statistics

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [❌]
| New feature/enhancement? (use the a.x branch)      | [✅]
| Deprecations?                          | [❌]
| BC breaks? (use the c.x branch)        | [❌]
| Automated tests included?              | [✅] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
This PR introduces a map preview with email read and click statistics. Statistics are consistent with existing charts - a read or click on an e-mail is counted as 1, regardless of the number of reopenings or the number of clicked links. The map legend shows the total number of reads/clicks and how many of them were made by contacts with a non-empty country, because the statistics for leads without a location cannot be marked on the map.

Email preview (s/emails/view/{email_id})

![map_email](https://github.com/mautic/mautic/assets/102536220/6827097b-971a-4b89-9ca5-6652247bd2b3)


Campaign preview (/s/campaigns/view/{campaign_id})
![campaign](https://github.com/mautic/mautic/assets/102536220/624c3aa6-1dea-4d34-bd28-2fa9c303076a)
Email campaign
![campaign_email](https://github.com/mautic/mautic/assets/102536220/5007bb92-0b12-4a8f-8cd0-d614a9a37bbc)

As part of this PR, zoom controls and map tooltip have been enlarged, so the Map widget on the dashboard and the map in the contact view (Places tab) will also change slightly in appearance, but their functionality should remain the same:
![dashboard](https://github.com/mautic/mautic/assets/102536220/029f9bb2-0d2f-4035-9d70-f183a2b8a5c2)


#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create at least one email. Send it to several contacts (preferably with a non-empty country field), then open and click on the link of at least one of messages.
3. Email statistics should be correctly displayed in the e-mail view (s/emails/view/{email_id}) "Map" tab. 
4. Create and run a test campaign with at least one "email.send" event.
5. Simulate reads and clicks on messages sent from the campaign.
6. Statistics from all emails sent from the campaign should be displayed in campaign preview (/s/campaigns/view/{campaign_id}) in "Map" tab.
